hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/rsql/VirtualPropertyResolverTest.java

```diff
@@ -0,0 +1,123 @@
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+
+import java.time.Instant;
+
+import org.apache.commons.lang3.text.StrSubstitutor;
+import org.eclipse.hawkbit.repository.TenantConfigurationManagement;
+import org.eclipse.hawkbit.repository.jpa.TimestampCalculator;
+import org.eclipse.hawkbit.repository.model.TenantConfigurationValue;
+import org.eclipse.hawkbit.tenancy.configuration.TenantConfigurationKey;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import ru.yandex.qatools.allure.annotations.Description;
+import ru.yandex.qatools.allure.annotations.Features;
+import ru.yandex.qatools.allure.annotations.Stories;
+
+@Features("Unit Tests - Repository")
+@Stories("Placeholder resolution for virtual properties")
+@RunWith(MockitoJUnitRunner.class)
+public class VirtualPropertyResolverTest {
+
+    @Spy
+    VirtualPropertyResolver resolverUnderTest = new VirtualPropertyResolver(new TimestampCalculator());
+
+    @Mock
+    TenantConfigurationManagement confMgmt;
+
+    StrSubstitutor substitutor;
+
+    Long nowTestTime;
+
+    private static final TenantConfigurationValue<String> TEST_POLLING_TIME_INTERVAL = TenantConfigurationValue
+            .<String> builder().value("00:05:00").build();
+    private static final TenantConfigurationValue<String> TEST_POLLING_OVERDUE_TIME_INTERVAL = TenantConfigurationValue
+            .<String> builder().value("00:07:37").build();
+
+    @Before
+    public void before() {
+        nowTestTime = Instant.now().toEpochMilli();
+        when(confMgmt.getConfigurationValue(TenantConfigurationKey.POLLING_TIME_INTERVAL, String.class))
+            .thenReturn(TEST_POLLING_TIME_INTERVAL);
+        when(confMgmt.getConfigurationValue(TenantConfigurationKey.POLLING_OVERDUE_TIME_INTERVAL, String.class))
+            .thenReturn(TEST_POLLING_OVERDUE_TIME_INTERVAL);
+
+        when(resolverUnderTest.getTimestampCalculator()).thenReturn(new TimestampCalculator() {
+            @Override
+            protected TenantConfigurationManagement getTenantConfigurationManagement() {
+                return confMgmt;
+            }
+        });
+
+        this.substitutor = new StrSubstitutor(resolverUnderTest,
+                StrSubstitutor.DEFAULT_PREFIX,
+                StrSubstitutor.DEFAULT_SUFFIX, StrSubstitutor.DEFAULT_ESCAPE);
+     }
+
+    @Test
+    @Description("Tests resolution of NOW_TS by using a StrSubstitutor configured with the VirtualPropertyResolver.")
+    public void resolveNowTimestampPlaceholder() {
+        String placeholder = "${NOW_TS}";
+        String testString = "lhs=lt=" + placeholder;
+
+        String resolvedPlaceholders = substitutor.replace(testString);
+        assertFalse("NOW_TS has to be resolved!", resolvedPlaceholders.contains(placeholder));
```

Comment:
Avoid the use of `assertFalse`, it is not easy to figure out for what you're asserting in the first place. Better use matchers:

`assertThat("NOW_TS has to be resolved!", resolvedPlaceholders,  Matchers.containsString(placeholder));`

If you'll use a static import for `Matchers` you'll don't need to provide the full qualified name